client/src/api.ts

```diff
@@ -158,6 +159,68 @@ export async function fetchSubjects(
   return result.data.data.subjects;
 }
 
+export async function fetchUsers(
+  searchParams?: SearchParams
+): Promise<Connection<User>> {
+  const params = { ...defaultSearchParams, ...searchParams };
+  const { filter, limit, cursor, sortBy, sortAscending } = params;
+  const result = await graphqlRequest.post("", {
+    query: `
+    query FetchUsers($filter: Object!, $limit: Int!, $cursor: String!, $sortBy: String!, $sortAscending: Boolean!){
+      fetchUsers(filter: $filter, limit: $limit,cursor: $cursor,sortBy: $sortBy,sortAscending: $sortAscending){
+        edges {
+          node {
+            _id
+            name
+            email
+            userRole
+            defaultMentor{
+              _id
+            }
+          }
+        }
+        pageInfo {
+          startCursor
+          endCursor
+          hasPreviousPage
+          hasNextPage
+        }
+      }
+    }`,
+    variables: {
+      filter: JSON.stringify(filter),
+      limit,
+      cursor,
+      sortBy,
+      sortAscending,
+    },
+  });
+  return result.data.data;
```

Comment:
don't assume GQL requests succeed. You have to check for errors, we should probably have a generic helper function for this